src/realm/object-store/sync/sync_session.cpp

```diff
@@ -870,6 +870,9 @@ void SyncSession::create_sync_session()
     session_config.proxy_config = sync_config.proxy_config;
     session_config.simulate_integration_error = sync_config.simulate_integration_error;
     session_config.flx_bootstrap_batch_size_bytes = sync_config.flx_bootstrap_batch_size_bytes;
+    session_config.session_reason = ClientResetOperation::is_fresh_path(m_config.path)
+                                        ? sync::SessionReason::ClientReset
+                                        : sync::SessionReason::Sync;
```

Comment:
We could do that, but we depend on when the session becomes active and sync::Session is created (we need to have the value set before that, otherwise it's too late). Ideally, this value would be set/inferred in the constructor of the SyncSession, but we need to pass the value through several layers. I can make the change if you feel strongly about it.

Reply:
OK - given our current session structure, looking for the path suffix seems like the most efficient approach, and `get_fresh_path_for()` is only used during `download_fresh_realm()`.